### PR TITLE
Update round_robin.go

### PR DESCRIPTION
**Pull request by loglogpro**
when n=MAX_UNIT32+1 = 0,  slice panic index of range.

and n-1 is no need.

### CODE REVIEW

File: round_robin.go [hunk 1]
```diff
@@ -33,5 +33,5 @@ func New(urls ...*url.URL) (RoundRobin, error) {
 // Next returns next address
 func (r *roundrobin) Next() *url.URL {
 	n := atomic.AddUint32(&r.next, 1)
-	return r.urls[(int(n)-1)%len(r.urls)]
+	return r.urls[(int(n)%len(r.urls)]
```
Comment (hlts2) [inline round_robin.go:36]:
```suggestion
	return r.urls[int(n)%len(r.urls)]
```

**Comment by hlts2** [comment]
Thank you for your PR 🙏 
I added one comment, please check it out.

File: round_robin.go (hunk 1 above)
Comment (loglogpro) [reply round_robin.go:36]: oh,sorry I mad a mistake with a left (. O(∩_∩)O

Comment (hlts2) [inline round_robin.go:36]: By the way, this implementation sets the initial value of the index to 1 and the test fails, so how about the following implementation?

```suggestion
	n := atomic.AddUint32(&r.next, 1) - 1
	if n < 0 {
		n = 0
	}
	return r.urls[int(n)%len(r.urls)]
```

Comment (loglogpro) [reply round_robin.go:36]: It's ok when you considering the test case. I use `n := atomic.AddUint32(&r.next, 1) return r.urls[int(n)%len(r.urls)]` in my project.

Appreciated your project.

**Pull request closed without merging**